### PR TITLE
Small updates to slice, str, io docs

### PR DESCRIPTION
**Pull request by bluss**
Main motivation was to update docs for the removal or "demotion" of certain extension traits. The update to the slice docs was larger, since the text was largely outdated.

### CODE REVIEW

**Comment by bluss** [comment]
Point of discussion: Instead of saying `StrExt::split` it now refers to `str::split`. This is actually UFCS-callable which is really nice. The type vs module confusion might otherwise make it a bit confusing, but it's meant to be read as "the method split on type str."

File: src/libcollections/slice.rs [hunk 1]
```diff
@@ -40,49 +39,44 @@
...
+//! for n in numbers {
+//!     println!("{} is a number!", n);
 //! }
 //! ```
 //!
-//! * `.iter_mut()` returns an iterator that allows modifying each value.
-//! * Further iterators exist that split, chunk or permute the slice.
+//! The mutable slice yields mutable references to the elements:
+//!
+//! ```rust
```
Comment (steveklabnik) [inline src/libcollections/slice.rs]: covention is to not include `rust` in API docs like this. I know it's strewn about with the rest of the file, but we shouldn't introduce new ones

File: src/libcollections/slice.rs [hunk 2]
```diff
@@ -40,49 +39,44 @@
...
+//! let mut scores = [7, 8, 9];
+//! for score in &mut scores[..] {
+//!     *score += 1;
+//! }
+//! ```
+//!
+//! The iterator yields references to the slice's elements, so if the element
+//! type of the slice is `i32`, the element type of the iterator is `&i32`.
```
Comment (steveklabnik) [inline src/libcollections/slice.rs]: Wouldn't these be `&mut i32`?

File: src/libcollections/slice.rs [hunk 3]
```diff
@@ -40,49 +39,44 @@
...
+//!
+//!
```
Comment (steveklabnik) [inline src/libcollections/slice.rs]: extra newline

File: src/libcore/str/mod.rs [hunk 4]
```diff
@@ -147,8 +147,8 @@ impl FromStr for bool {
     /// assert!(<bool as FromStr>::from_str("not even a boolean").is_err());
     /// ```
     ///
-    /// Note, in many cases, the StrExt::parse() which is based on
-    /// this FromStr::from_str() is more proper.
+    /// Note, in many cases, the `str::parse()` which is based on
+    /// this `FromStr::from_str()` is more proper.
```
Comment (steveklabnik) [inline src/libcore/str/mod.rs]: I know you didn't write this sentence, but while you're here, maybe update it? It sounds akward to me:

> Note, in many cases, the str::parse() function is more proper.

**Comment by steveklabnik** [comment]
Thank you! this is pretty great, r=me after we handle the nits

**Comment by nikomatsakis** [comment]
(I was just changing the assignment)

**Comment by bluss** [comment]
Thank you for the review, fixed the nits! I changed the parse sentence to call it a method though.

**Comment by steveklabnik** [comment]
Would you mind squashing? sorry

**Comment by bluss** [comment]
Squashed into three commits again.

**Comment by bluss** [comment]
Rebased to new master.

**Comment by Manishearth** [comment]
@bors: r=steveklabnik rollup

Looks like everything was addressed. Thanks!

**Pull request merged**